packages/cli/src/commands/list/index.ts

```diff
@@ -94,10 +98,10 @@ export default async function list(client: Client) {
   });
 
   let project: Project;
-  let pagination;
+  let pagination: PaginationOptions | undefined;
```

Comment:
Is `let variableName;` a pattern we actually want to avoid?

While typescript says the value of pagination is `any` at declaration, the inference is really good wherever it is used. For example, if you try to use pagination before it has a chance of being set, typescript knows it can only be `undefined`. If we type it `PaginationOptions | undefined`, we lose that strictness. Also, the tsconfig rule `noImplicitAny` doesn't actually complain about this case.

Reply:
Not a blocking comment, just curious about what to prefer for future changes.

Reply:
Yes, I would like to avoid this construction where possible, and instantiating the type where it's not. The problem with inference is that it will automatically follow what we add to the variable, and the implicit `any` means that we won't necessarily get an error from a mistaken type change, unless it causes another issue in the later inferred code.

Reply:
I'm not certain that I fully understand type inference in Typescript, but:

```typescript
let pagination: PaginationOptions | undefined = undefined;
```

should both prevent the implicit any, and allow for proper inference.

Reply:
Good point. Also, just found an example that makes me really dislike the `let varName;` pattern:

https://www.typescriptlang.org/play/?ssl=10&ssc=39&pln=1&pc=1#code/DYUwLgBAhg3AUHKEC8EAs84DMCuA7AYzAEsB7PCLU0gCgEoIBvOCaFCAIgCMoAnD+AF9s1eplgQA9JLbEAzhDABPAA4gIeHAFsuIXgBoNpSHLC9ieAOYQAhHaA

Seems like types can become totally incorrect with closure mutation! I'm all for avoiding `let variableName;` now